Assets/Plugins/Editor/Uplift/Schemas/Repositories/GithubRepository.cs

```
@@ -0,0 +1,155 @@
+            foreach(GitHubAsset asset in upsetAssets)
+            {
+                StrictXmlDeserializer<Upset> deserializer = new StrictXmlDeserializer<Upset>();
+                
+                EditorUtility.DisplayProgressBar(
+                    progressBarTitle,
+                    "Parsing " + asset.name, 
+                    (float)(index++) / upsetAssets.Length
+                );
+                
+                using(StreamReader sr = new StreamReader(GitHub.GetAssetStream(asset, GetToken())))
+                {
+                    Upset upset = deserializer.Deserialize(sr.BaseStream);
+                    upset.MetaInformation.dirName = asset.name;
+                    upsetList.Add(upset);
+                }
+            }
+
+            EditorUtility.ClearProgressBar();
+            return upsetList.ToArray();
+        }
+
+        private GitHubRelease[] GetPackagesReleases()
+        {
+            if(releases == null)
+            {
+                IEnumerator e = GitHub.LoadReleases(urlField, GetToken());
+                do { Thread.Sleep(1000); } while (e.MoveNext());
+
+                GitHubRelease[] fetchedReleases = (GitHubRelease[]) e.Current;
+
+                if (fetchedReleases == null)
+                    throw new ApplicationException("This github repository does not have releases");
+
+                string[] tagArray;
+                if(tagListField == null || tagListField.Length == 0)
+                {
+                    tagArray = new string[] { "packages" };
+                }
+                else
+                {
+                    tagArray = tagListField;
+                    Array.Resize<string>(ref tagArray, tagListField.Length + 1);
+                    tagArray[tagListField.Length] = "packages";
+                }
+
+                releases = fetchedReleases.Where(rel => tagArray.Contains(rel.tag)).ToArray();
+                
+                if (releases == null)
```

Can never be null. Could be empty though. But given the above, I guess we want to fail when we don't find the ones that are specified.

---

I implemented it the following way:
- Fail hard if no release is found (likely repo misconfiguration)
- Only log if partially found